pkg/common/resources/resources_test.go

```diff
@@ -1701,3 +1701,29 @@ func TestIsEmpty(t *testing.T) {
 		})
 	}
 }
+
+func TestResource_DominantResource(t *testing.T) {
+	tests := []struct {
+		name     string
+		used     *Resource
+		capacity *Resource
+		wantName string
+	}{
+		{"nil receiver", nil, Zero, ""},
+		{"nil cap", Zero, nil, ""},
+		{"zero cap", NewResourceFromMap(map[string]Quantity{"A": 10}), Zero, ""},
+		{"over cap", NewResourceFromMap(map[string]Quantity{"A": 20}), NewResourceFromMap(map[string]Quantity{"A": 10}), "A"},
+		{"zero usage exist", NewResourceFromMap(map[string]Quantity{"A": 0}), NewResourceFromMap(map[string]Quantity{"A": 10}), "A"},
+		{"usage not in cap", NewResourceFromMap(map[string]Quantity{"B": 10}), NewResourceFromMap(map[string]Quantity{"A": 10}), ""},
+		{"multiple usages", NewResourceFromMap(map[string]Quantity{"B": 10, "A": 10}), NewResourceFromMap(map[string]Quantity{"A": 10, "B": 20}), "A"},
+		{"0 usage with 0 cap", NewResourceFromMap(map[string]Quantity{"A": 0}), NewResourceFromMap(map[string]Quantity{"A": 0}), "A"},
```

Comment:
Change it to "" to pass the test:
`{"0 usage with 0 cap", NewResourceFromMap(map[string]Quantity{"A": 0}), NewResourceFromMap(map[string]Quantity{"A": 0}), ""},`
Or we handle the 0/0 situation in `DominantResourceType`

Reply:
See below special cased both values 0 case